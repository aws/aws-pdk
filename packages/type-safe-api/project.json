{
  "name": "@aws/type-safe-api",
  "root": "packages/type-safe-api",
  "targets": {
    "build": {
      "inputs": [
        "default",
        "^default",
        "!{projectRoot}/.jsii",
        "!{projectRoot}/lib/**/*",
        "!{projectRoot}/dist/**/*",
        "!{projectRoot}/coverage/**/*",
        "!{projectRoot}/test-reports/**/*",
        "!{projectRoot}/docs/api/**/*"
      ],
      "outputs": [
        "{projectRoot}/.jsii",
        "{projectRoot}/lib",
        "{projectRoot}/dist",
        "{projectRoot}/coverage",
        "{projectRoot}/test-reports",
        "{projectRoot}/docs/api"
      ],
      "dependsOn": [
        "^build"
      ],
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen build",
        "cwd": "packages/type-safe-api"
      }
    },
    "default": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen default",
        "cwd": "packages/type-safe-api"
      }
    },
    "pre-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen pre-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "post-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen post-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test",
        "cwd": "packages/type-safe-api"
      }
    },
    "package": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package",
        "cwd": "packages/type-safe-api"
      }
    },
    "test:watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test:watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint",
        "cwd": "packages/type-safe-api"
      }
    },
    "compat": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compat",
        "cwd": "packages/type-safe-api"
      }
    },
    "package-all": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package-all",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:java": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:java",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:python": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:python",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint-staged": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint-staged",
        "cwd": "packages/type-safe-api"
      }
    },
    "jest": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen jest",
        "cwd": "packages/type-safe-api"
      }
    },
    "docgen": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen docgen",
        "cwd": "packages/type-safe-api"
      }
    }
  },
  "implicitDependencies": [
    "smithy-async-transformer"
  ],
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"pnpm exec projen\"."
}
