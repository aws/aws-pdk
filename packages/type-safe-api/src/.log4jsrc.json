{
    "appenders": {
        "out": {
            "type": "stdout",
            "layout": {
                "type": "pattern",
                "pattern": "%[[%d{yyyy-MM-dd-hh:mm:ss}] [%p] [%f{1}] -%] %m"
            }
        }
    },
    "categories": {
        "default": { "appenders": [ "out" ], "level": "INFO", "enableCallStack": true }
    }
}