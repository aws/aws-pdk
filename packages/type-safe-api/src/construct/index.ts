/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./type-safe-rest-api";
export * from "./waf/types";
export * from "./authorizers";
export * from "./integrations";
export * from "./spec";
export * from "./functions";
