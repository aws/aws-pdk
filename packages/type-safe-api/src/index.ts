/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
// instantiate log4js when app is initialized
import { configure as configureLog4js } from "log4js";
import * as log4jsSettings from "./.log4jsrc.json";

configureLog4js(log4jsSettings);

export * from "./construct";
export * from "./project";
