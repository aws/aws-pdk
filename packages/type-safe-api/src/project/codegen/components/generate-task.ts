/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { ProjectUtils } from "@aws/monorepo";
import { Component, Project, Task } from "projen";

/**
 * Component which manages a "generate" task for a project
 */
export class GenerateTask extends Component {
  /**
   * Retrieves an instance of GenerateTask if one is associated to the given project.
   *
   * @param project project instance.
   */
  static of(project: Project): Task | undefined {
    return (
      project.components.find((c) =>
        ProjectUtils.isNamedInstanceOf(c, GenerateTask)
      ) as GenerateTask | undefined
    )?.task;
  }

  /**
   * Retrieves an instance of GenerateTask if one is associated to the given project,
   * otherwise creates a GenerateTask instance for the project.
   *
   * @param project project instance.
   */
  static ensure(project: Project): Task {
    return GenerateTask.of(project) || new GenerateTask(project).task;
  }

  public readonly task: Task;

  constructor(project: Project) {
    super(project);

    this.task =
      project.tasks.tryFind("generate") ?? project.addTask("generate");
  }
}
