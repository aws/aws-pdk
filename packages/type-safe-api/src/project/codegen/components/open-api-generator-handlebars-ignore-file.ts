/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { IgnoreFile, Project } from "projen";
import { OpenApiGeneratorIgnoreFile } from "./open-api-generator-ignore-file";

/**
 * Represents an .openapi-generator-ignore-handlebars file.
 * This is used for an optional second code generation pass to generate files using the handlebars templating engine
 * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/customization.md#ignore-file-format
 */
export class OpenApiGeneratorHandlebarsIgnoreFile extends IgnoreFile {
  constructor(project: Project) {
    super(project, ".openapi-generator-ignore-handlebars");

    // Ignore all files by default to allow opting in to generate with specific handlebars templates
    this.addPatterns(...OpenApiGeneratorIgnoreFile.ALL_FILES_PATTERNS);
  }
}
