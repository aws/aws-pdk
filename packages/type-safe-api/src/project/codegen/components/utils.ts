/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import * as path from "path";
import * as readPkg from "read-pkg-up";
import { Language, Library } from "../../languages";
import { MockResponseDataGenerationOptions } from "../../types";

/**
 * Enum for generator directories for non-runtime generators
 */
export enum OtherGenerators {
  DOCS = "docs",
  // Infrastructure
  TYPESCRIPT_CDK_INFRASTRUCTURE = "typescript-cdk-infrastructure",
  PYTHON_CDK_INFRASTRUCTURE = "python-cdk-infrastructure",
  JAVA_CDK_INFRASTRUCTURE = "java-cdk-infrastructure",
  // Handlers
  TYPESCRIPT_LAMBDA_HANDLERS = "typescript-lambda-handlers",
  PYTHON_LAMBDA_HANDLERS = "python-lambda-handlers",
  JAVA_LAMBDA_HANDLERS = "java-lambda-handlers",
}

export enum TypeSafeApiScript {
  PARSE_OPENAPI_SPEC = "parse-openapi-spec",
  GENERATE = "generate",
  GENERATE_MOCK_DATA = "generate-mock-data",
  GENERATE_HTML_REDOC_DOCS = "generate-html-redoc-docs",
  CLEAN_OPENAPI_GENERATED_CODE = "clean-openapi-generated-code",
  COPY_GRADLE_WRAPPER = "copy-gradle-wrapper",
}

/**
 * Generator directory for openapi generation containing templates, config etc.
 */
export type GeneratorDirectory = Language | Library | OtherGenerators;

/**
 * Types of normalizers supported by openapi-generator
 * @see https://openapi-generator.tech/docs/customization/#openapi-normalizer
 */
export type OpenApiNormalizer = "KEEP_ONLY_FIRST_TAG_IN_OPERATION";

/**
 * Options for generating code or docs using OpenAPI Generator CLI
 */
export interface GenerationOptions {
  /**
   * The OpenAPI generator to use to generate the code/docs
   */
  readonly generator: string;
  /**
   * The directory to use for OpenAPI generation
   */
  readonly generatorDirectory: GeneratorDirectory;
  /**
   * The path of the OpenAPI spec to generate code from
   */
  readonly specPath: string;
  /**
   * Additional properties to pass to the generate cli
   */
  readonly additionalProperties?: {
    [key: string]: string;
  };
  /**
   * Supply the relative path from the code project root to the source code directory in which custom generated files
   * (eg. operation config) should be placed.
   */
  readonly srcDir?: string;
  /**
   * Normalizers to apply to the spec prior to generation, if any
   * @see https://openapi-generator.tech/docs/customization/#openapi-normalizer
   */
  readonly normalizers?: Partial<Record<OpenApiNormalizer, boolean>>;
  /**
   * Vendor extensions to add for code generation, allowing custom properties to be passed to the generator templates
   * Keys should begin with "x-"
   */
  readonly extraVendorExtensions?: Record<string, string | boolean>;
  /**
   * Generate alias as model.
   * Defines whether "primitive types" defined at the model/schema level will be wrapped in a model (ie maps/lists)
   * @see https://openapi-generator.tech/docs/globals/
   * @default true
   */
  readonly generateAliasAsModel?: boolean;
  /**
   * The path to the json smithy model file, if available
   */
  readonly smithyJsonPath?: string;
}

/**
 * Build a command for running a script from this project's bin
 */
export const buildTypeSafeApiExecCommand = (
  script: TypeSafeApiScript,
  args?: string
) => {
  const { packageJson } = readPkg.sync({
    cwd: path.resolve(__dirname),
  })!;
  return `npx --yes -p aws-pdk@${packageJson.version} ${script}${
    args ? ` ${args}` : ""
  }`;
};

const serializeProperties = (properties: { [key: string]: string }) =>
  Object.entries(properties)
    .map(([key, value]) => `${key}=${value}`)
    .join(",");

/**
 * Generate code or docs by invoking the root generate script
 */
export const buildInvokeOpenApiGeneratorCommandArgs = (
  options: GenerationOptions
): string => {
  const srcDir = options.srcDir ?? "src";
  const additionalProperties = options.additionalProperties
    ? ` --additional-properties "${serializeProperties(
        options.additionalProperties
      )}"`
    : "";

  const normalizers = options.normalizers
    ? ` --openapi-normalizer "${serializeProperties(
        Object.fromEntries(
          Object.entries(options.normalizers).map(([k, v]) => [k, `${v}`])
        )
      )}"`
    : "";

  const extensions = options.extraVendorExtensions
    ? ` --extra-vendor-extensions '${JSON.stringify(
        options.extraVendorExtensions
      )}'`
    : "";

  const generateAliasAsModel =
    options.generateAliasAsModel ?? true ? " --generate-alias-as-model" : "";

  const smithyJsonPath = options.smithyJsonPath
    ? ` --smithy-json-path ${options.smithyJsonPath}`
    : "";

  const specPath = options.specPath;
  const outputPath = ".";

  return `--generator ${options.generator} --spec-path ${specPath} --output-path ${outputPath} --generator-dir ${options.generatorDirectory} --src-dir ${srcDir}${smithyJsonPath}${additionalProperties}${normalizers}${extensions}${generateAliasAsModel}`;
};

/**
 * Builds a command to clean up files which were previously generated by openapi generator
 */
export const buildCleanOpenApiGeneratedCodeCommand = (): string => {
  return buildTypeSafeApiExecCommand(
    TypeSafeApiScript.CLEAN_OPENAPI_GENERATED_CODE,
    `--code-path .`
  );
};

/**
 * Options for generating mock data json files
 */
export interface MockDataGenerationOptions
  extends MockResponseDataGenerationOptions {
  /**
   * The path of the OpenAPI spec to generate data for
   */
  readonly specPath: string;
  /**
   * Output sub directory relative to the outdir in which to generate mock data
   * Mock data will be written to a directory named 'mocks' within the sub directory
   * @default .
   */
  readonly outputSubDir?: string;
}

/**
 * Invoke the mock data generator script
 */
export const buildInvokeMockDataGeneratorCommand = (
  options: MockDataGenerationOptions
): string => {
  const outputPath = options.outputSubDir ?? ".";
  const locale = options.locale ? ` --locale ${options.locale}` : "";
  const maxArrayLength =
    options.maxArrayLength !== undefined
      ? ` --max-array-length ${options.maxArrayLength}`
      : "";
  return buildTypeSafeApiExecCommand(
    TypeSafeApiScript.GENERATE_MOCK_DATA,
    `--spec-path ${options.specPath} --output-path ${outputPath}${locale}${maxArrayLength}`
  );
};
