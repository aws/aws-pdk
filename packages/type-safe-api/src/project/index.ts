/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./model/type-safe-api-model-build";
export * from "./model/type-safe-api-async-model-build";
export * from "./model/smithy/smithy-project-definition";
export * from "./model/openapi/open-api-project-definition";
export * from "./type-safe-api-project";
export * from "./model/type-safe-api-model-project";
export * from "./model/smithy/smithy-definition";
export * from "./model/openapi/open-api-definition";
export * from "./type-safe-websocket-api-project";
export * from "./model/type-safe-websocket-api-model-project";
export * from "./model/smithy/smithy-async-definition";
export * from "./model/openapi/open-api-async-definition";
export * from "./model/smithy/types";
export * from "./model/openapi/open-api-model-project";
export * from "./model/openapi/open-api-async-model-project";
export * from "./model/smithy/smithy-async-model-project";
export * from "./model/smithy/smithy-model-project";
export * from "./languages";
export * from "./types";
export * from "./typescript-project-options";
export * from "./python-project-options";
export * from "./java-project-options";
