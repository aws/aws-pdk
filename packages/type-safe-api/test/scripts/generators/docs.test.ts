/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import os from "os";
import * as path from "path";
import { exec } from "projen/lib/util";
import { buildCodegenCommandArgs } from "../../../lib/project/codegen/components/utils";
import { DocumentationFormat } from "../../../src";
import { withTmpDirSnapshot } from "../../project/snapshot-utils";

describe("Docs Generation Script Unit Tests", () => {
  it.each([DocumentationFormat.MARKDOWN, DocumentationFormat.PLANTUML])(
    "Generates %s",
    (format) => {
      const specPath = path.resolve(
        __dirname,
        `../../resources/specs/single.yaml`
      );

      expect(
        withTmpDirSnapshot(
          os.tmpdir(),
          (outdir) => {
            exec(`cp ${specPath} ${outdir}/spec.yaml`, {
              cwd: path.resolve(__dirname),
            });
            exec(
              `${path.resolve(
                __dirname,
                "../../../scripts/type-safe-api/run.js generate"
              )} ${buildCodegenCommandArgs({
                specPath: "spec.yaml",
                templateDirs: [`docs/templates/${format}`],
              })}`,
              {
                cwd: outdir,
              }
            );
          },
          { excludeGlobs: ["spec.yaml"] }
        )
      ).toMatchSnapshot();
    }
  );
});
