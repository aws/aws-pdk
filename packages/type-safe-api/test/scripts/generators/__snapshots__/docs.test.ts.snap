// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Docs Generation Script Unit Tests Generates markdown 1`] = `
{
  ".tsapi-manifest": "README.md
Apis/DefaultApi.md
Models/ApiError.md
Models/MapResponse.md
Models/MapResponseMapPropertyValue.md
Models/TestRequest.md
Models/TestResponse.md
Models/TestResponseMessagesInner.md",
  "Apis/DefaultApi.md": "# DefaultApi
    

| Method | HTTP request | Description |
|------------- | ------------- | -------------|
| [**anyRequestResponse**](DefaultApi.md#anyRequestResponse) | **PUT** /any-request-response |  |
| [**empty**](DefaultApi.md#empty) | **PUT** /empty-response |  |
| [**mapResponse**](DefaultApi.md#mapResponse) | **GET** /map-response |  |
| [**mediaTypes**](DefaultApi.md#mediaTypes) | **POST** /different-media-type |  |
| [**multipleContentTypes**](DefaultApi.md#multipleContentTypes) | **POST** /multiple-content-types |  |
| [**operationOne**](DefaultApi.md#operationOne) | **POST** /path/{pathParam} |  |
| [**withoutOperationIdDelete**](DefaultApi.md#withoutOperationIdDelete) | **DELETE** /without-operation-id |  |

<a name="anyRequestResponse"></a>
# **anyRequestResponse**
> Object anyRequestResponse(body)


### Parameters
|Name | Type | Description  | Notes |
|------------- | ------------- | ------------- | -------------|
| **body** | **Object**|  | [optional] |

### Return type

**Object**

### HTTP request headers

- **Content-Type**: application/json
- **Accept**: application/json

<a name="empty"></a>
# **empty**
> empty()


### Parameters
This endpoint does not need any parameters.

### Return type

null (empty response body)

### HTTP request headers

- **Content-Type**: Not defined
- **Accept**: Not defined

<a name="mapResponse"></a>
# **mapResponse**
> MapResponse mapResponse()


### Parameters
This endpoint does not need any parameters.

### Return type

[**MapResponse**](../Models/MapResponse.md)

### HTTP request headers

- **Content-Type**: Not defined
- **Accept**: application/json

<a name="mediaTypes"></a>
# **mediaTypes**
> String mediaTypes(body)


### Parameters
|Name | Type | Description  | Notes |
|------------- | ------------- | ------------- | -------------|
| **body** | **byte[]**|  | |

### Return type

**String**

### HTTP request headers

- **Content-Type**: application/pdf
- **Accept**: application/x-mpegurl

<a name="multipleContentTypes"></a>
# **multipleContentTypes**
> String multipleContentTypes(body)


### Parameters
|Name | Type | Description  | Notes |
|------------- | ------------- | ------------- | -------------|
| **body** | [**TestRequest**](../Models/TestRequest.md)|  | |

### Return type

**String**

### HTTP request headers

- **Content-Type**: application/json, application/pdf
- **Accept**: application/json

<a name="operationOne"></a>
# **operationOne**
> TestResponse operationOne(param1, param2, param3, pathParam, x-header-param, body, param4, x-multi-value-header-param)


### Parameters
|Name | Type | Description  | Notes |
|------------- | ------------- | ------------- | -------------|
| **param1** | **String**| This is parameter 1 | |
| **param2** | **List<String>**| This is parameter 2 | |
| **param3** | **BigDecimal**|  | |
| **pathParam** | **String**|  | |
| **x-header-param** | **String**| This is a header parameter | |
| **body** | [**TestRequest**](../Models/TestRequest.md)|  | |
| **param4** | **String**|  | [optional] |
| **x-multi-value-header-param** | **List<String>**|  | [optional] |

### Return type

[**TestResponse**](../Models/TestResponse.md)

### HTTP request headers

- **Content-Type**: application/json
- **Accept**: application/json

<a name="withoutOperationIdDelete"></a>
# **withoutOperationIdDelete**
> TestResponse withoutOperationIdDelete()


### Parameters
This endpoint does not need any parameters.

### Return type

[**TestResponse**](../Models/TestResponse.md)

### HTTP request headers

- **Content-Type**: Not defined
- **Accept**: application/json

",
  "Models/ApiError.md": "# ApiError
## Properties

| Name | Type | Description | Notes |
|------------ | ------------- | ------------- | -------------|
| **errorMessage** | **String** |  |  |

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

",
  "Models/MapResponse.md": "# MapResponse
## Properties

| Name | Type | Description | Notes |
|------------ | ------------- | ------------- | -------------|
| **mapProperty** | **Map<String, MapResponseMapPropertyValue>** |  |  |

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

",
  "Models/MapResponseMapPropertyValue.md": "# MapResponseMapPropertyValue
## Properties

| Name | Type | Description | Notes |
|------------ | ------------- | ------------- | -------------|
| **a** | **String** |  |  |
| **b** | **String** |  | [optional]  |

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

",
  "Models/TestRequest.md": "# TestRequest
## Properties

| Name | Type | Description | Notes |
|------------ | ------------- | ------------- | -------------|
| **myInput** | **BigDecimal** |  | [optional]  |

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

",
  "Models/TestResponse.md": "# TestResponse
## Properties

| Name | Type | Description | Notes |
|------------ | ------------- | ------------- | -------------|
| **messages** | **List<TestResponseMessagesInner>** |  |  |

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

",
  "Models/TestResponseMessagesInner.md": "# TestResponseMessagesInner
## Properties

| Name | Type | Description | Notes |
|------------ | ------------- | ------------- | -------------|
| **message** | **String** | This is a message | [optional]  |
| **id** | **Integer** |  |  |

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)

",
  "README.md": "# Documentation for Example API

<a name="documentation-for-api-endpoints"></a>
## Documentation for API Endpoints

| Class | Method | HTTP request | Description |
|------------ | ------------- | ------------- | -------------|
| *DefaultApi* | [**anyRequestResponse**](Apis/DefaultApi.md#anyRequestResponse) | **PUT** /any-request-response |  |
| *DefaultApi* | [**empty**](Apis/DefaultApi.md#empty) | **PUT** /empty-response |  |
| *DefaultApi* | [**mapResponse**](Apis/DefaultApi.md#mapResponse) | **GET** /map-response |  |
| *DefaultApi* | [**mediaTypes**](Apis/DefaultApi.md#mediaTypes) | **POST** /different-media-type |  |
| *DefaultApi* | [**multipleContentTypes**](Apis/DefaultApi.md#multipleContentTypes) | **POST** /multiple-content-types |  |
| *DefaultApi* | [**operationOne**](Apis/DefaultApi.md#operationOne) | **POST** /path/{pathParam} |  |
| *DefaultApi* | [**withoutOperationIdDelete**](Apis/DefaultApi.md#withoutOperationIdDelete) | **DELETE** /without-operation-id |  |

<a name="documentation-for-models"></a>
## Documentation for Models

 - [ApiError](./Models/ApiError.md)
 - [MapResponse](./Models/MapResponse.md)
 - [MapResponseMapPropertyValue](./Models/MapResponseMapPropertyValue.md)
 - [TestRequest](./Models/TestRequest.md)
 - [TestResponse](./Models/TestResponse.md)
 - [TestResponseMessagesInner](./Models/TestResponseMessagesInner.md)
",
}
`;

exports[`Docs Generation Script Unit Tests Generates plantuml 1`] = `
{
  ".tsapi-manifest": "schemas.plantuml",
  "schemas.plantuml": "@startuml

title Example API Schemas Diagram

entity ApiError {
    * errorMessage: String
}

entity MapResponse {
    * mapProperty: Map<String, MapResponseMapPropertyValue>
}

entity MapResponseMapPropertyValue {
    * a: String
    b: String
}

entity TestRequest {
    myInput: BigDecimal
}

entity TestResponse {
    * messages: List<TestResponseMessagesInner>
}

entity TestResponseMessagesInner {
    message: String
    * id: Integer
}


MapResponse -- "0..*" MapResponseMapPropertyValue : mapProperty
TestResponse -- "0..*" TestResponseMessagesInner : messages

@enduml",
}
`;
